muspinsim/__main__.py

```diff
@@ -24,29 +24,73 @@
 LOGFORMAT = "[%(levelname)s] [%(threadName)s] [%(asctime)s] %(message)s"
 
 
+def check_dir_path(string):
+    if os.path.isdir(string):
+        return string
+    else:
+        try:
+            os.makedirs(string)
+        except OSError:
+            raise NotADirectoryError(string)
+    return string
+
+
 def main(use_mpi=False):
 
     if use_mpi:
         mpi.connect()
 
     if mpi.is_root:
         # Entry point for script
-        parser = ap.ArgumentParser()
+        parser = ap.ArgumentParser(
+            description="Muspinsim - A program designed to carry out "
+            "spin dynamics calculations for muon science experiments"
+        )
         parser.add_argument(
-            "input_file",
+            "-o",
```

Comment:
Another common convention is to have optional arguments in alphabetical order (aside from `-h` which is sometimes kept at the top). Makes it easier if you're looking for something specific in the help section.